### PR TITLE
Add TLS Support for MQTT

### PR DESCRIPTION
**Pull request by ronytomen**
# Description

Add TLS support for MQTT.  Not completely sure this is the correct way to implement this for MQTT, but I followed other examples I found that worked. Comments welcomed, what can I do better!

Fixes # N/A

## Type of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: mqtt.py [hunk 1]
```diff
@@ -51,6 +56,20 @@ def username(self):
   def password(self):
     return self._config['password'] if 'password' in self._config else None
 
+  @property
+  def ca_cert(self):
+    return self._config['ca_cert'] if 'ca_cert' in self._config else None
+
+  @property
+  def ca_verify(self):
```
Comment (zewelor) [inline mqtt.py]: Can this be simplified ? For example

```
if 'ca_verify' in self._config:
  return self._config['ca_verify']
else
  return True
```

Comment (ronytomen) [reply mqtt.py]: I went back and forth.  I forced true/false to filter invalid user input in the configuration file.  Maybe it should care?

Remove it?  Keep it, but add comment to clarify?

Comment (zewelor) [reply mqtt.py]: Comment would be good :D

Comment (ronytomen) [reply mqtt.py]: Comment added

**Comment by zewelor** [comment]
Great merging thanks

**Pull request merged**